library/common-library/DEVELOPERS.md

```diff
@@ -99,6 +99,37 @@ spec:
 
 
 
+## _annotations.tpl
+### `newrelic.common.annotations.deployment` and `newrelic.common.annotations.pod`
+These are functions that are used to annotate objects. They are configured by this `values.yaml`
+```yaml
+global:
+  deploymentAnnotations: {}  # included in all the deployments and daemonsets of all the charts
+  podAnnotations: {}  # included in all the pods of all the charts
```

Comment:
why for labels we have pod and generic labels and for annotations we have merely pod and deployment?

Reply:
Because `labels` are labeled for **all** objects while annotations are only for things that deploy pods. Would you want me to change it to something like `deployerAnnotations`?

Reply:
Having both DeploymentAnnotations and DaemonSet annotations seems the obvious solution, but annoying logic would be needed in the control Plane component

Reply:
I think this is a very convoluted use case 🤔 
If we're really sure we want to implement this i'd go with

```
workloads:
  annotaitons:
```

Just as we do with e.g. `serviceAccount`.

Reply:
We have a `deployment.annotations` in `nri-kube-events`: https://github.com/newrelic/helm-charts/pull/794/files#diff-8de0ccfb5402d7bfb33702ff03f10bd570021abbfefb098c89c3f2932a75a729L118-L120

So we might want to follow this:
```yaml
deployment:
  annotations: {}
daemonset:
  annotations: {}
statefulset:
  annotations: {}
```

But, to be honest, there is no option that I like. I dislike less to simply put `deploymentAnnotations` and use it for Damonsets and StatefulSets...

Reply:
As we decided offline, this feature is out of scope of the PR and the way that we like could take more time that the timebox we have to solve all the issues we have.

Reverted in [#5c4233b](https://github.com/newrelic/helm-charts/pull/802/commits/5c4233b1f6e9d72e2d18e475d5bb705fad8f4e88)